content/deyemiobaa/creating-queues-using-javascript.md

```diff
@@ -0,0 +1,229 @@
+Categories:
+  - "computer-science"
+Tags:
+  - "Algorithms"
+  - "Arrays"
+  - "ES6"
+  - "Classes"
+CatalogContent:
+  - "introduction-to-javascript"
+  - "linear-data-structures"
+---
+
+_**Prerequisites:** JavaScript_  
+_**Versions:** ECMAScript 2015 (ES6)_
+
+## Introduction
+
+During conversations about data structures in the programming world, the term queue is heard quite often. A queue is a linear data structure that follows the first-in-first-out (FIFO) pattern, i.e., removal takes place at the front, and addition takes place at the end. Think of it as the checkout point at a grocery store. Customers who get to the checkout counter first get attended to first. They can only join the queue from the back of the line, and after they get attended to, they leave from the front of the line.
+
+## Linear data structures
+
+Linear data structures are types of data structures in which elements are stored sequentially. The elements are organized in such a way that each element is directly linked to its previous and next element. However, the first element is only linked to its previous element and the last element is only linked to the next element.
+
+Queues follow this arrangement pattern, and we can see that by referring to our grocery store checkout counter analogy. Each customer that walks to the checkout counter represents an element. The first customer has a link to the previous customer, the last customer has a link to the next customer and every other customer in-between has a link to a previous and next customer.
+
+## Use cases for queues
+
+There are several use cases for queues in the programming and real world. Some of them include the following.
+
+### A desk printer
+
+When you send documents to the printer, they are printed in the same order in which they are sent, and this is very useful when you're trying to print documents that follow a specific order. The printer makes sure the pages don't get mixed up.
+
+### Adding songs to a queue in your music player
+
+Sometimes, while we are working on a task, we have our go-to songs to keep the momentum going. The queue function of the music player makes sure you listen to your song selections just the way it is arranged.
+
+### Customer service wait-lines
+  
+If you go to a bank to file a complaint, they make you wait in lines and the next person in line won't be called unless the issue of the current customer has been resolved.
+
+### File sharing/data transfer between two processes
+
+When transferring files from one device to another, the files are received on the other end in the same order they were sent, irrespective of the time it takes for any of them to get completed.
+
+## Implementation of Queues
+
+Queues can be implemented in any programming language, but our focus is on how to create them using JavaScript. In JavaScript, it can also be implemented in two ways: arrays and linked lists. For this article, we will implement queues using arrays.
+
```

Comment:
@Dusch4593, I've implemented the suggested changes and added one to this part of the article. The paragraph was looking isolated so I added a header to it.

I also wanted to attach links to these two terms, `array` and `linked list`, but I could only find an article for arrays on the Codecademy blog so I left it. If you have any suggestions or you think it's fine this way, please let me know. Thanks

Reply:
That's fine! 😄 We can leave this as it is.